dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -355,13 +363,12 @@ private static void printStaticLinkageError(
           String className = classInfo.getName();
           JavaClass javaClass = repository.loadClass(className);
           String topLevelClassName = javaClass.getClassName();
-          internalClassNames.add(topLevelClassName);
+          pathToClasses.put(jarFilePath, topLevelClassName);
           // This does not take double-nested classes. As long as such classes are accessed
           // only from the outer class, static linkage checker does not report false positives
```

Comment:
The missing methods will be reported. It's easy because constant pool of class `A` has all references to other classes (`Z.z`, `Q1.q1`, `Q2.q2`, `Y1.y1`, `Y1.y1`, `Y2.y2`).
From this example, there's no false positive, because they (`A` calling `A.B.b`, `A` calling `A.B.C.c` and `B` calling `A.B.C.c`) are all internal references. The tool assumes there is no linkage error within a jar file.

False positive would be a class `X` in different jar file calling `A.B.D.publicD()`.